extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java

```diff
@@ -166,8 +205,268 @@ public void build(SmallRyeReactiveMessagingRecorder recorder, BeanContainerBuild
+                ResultHandle containerHandle = invoke
+                        .invokeStaticMethod(MethodDescriptor.ofMethod(Arc.class, "container", ArcContainer.class));
+                ResultHandle beanHandle = invoke.invokeInterfaceMethod(
+                        MethodDescriptor.ofMethod(ArcContainer.class, "bean", InjectableBean.class, String.class),
+                        containerHandle, invoke.load(bean.getIdentifier()));
+                ResultHandle instanceHandle = invoke.invokeInterfaceMethod(
+                        MethodDescriptor.ofMethod(ArcContainer.class, "instance", InstanceHandle.class, InjectableBean.class),
+                        containerHandle, beanHandle);
+                ResultHandle beanInstanceHandle = invoke
+                        .invokeInterfaceMethod(MethodDescriptor.ofMethod(InstanceHandle.class, "get", Object.class),
+                                instanceHandle);
+
+                int parametersCount = method.parameters().size();
+                String[] argTypes = new String[parametersCount];
+                ResultHandle[] args = new ResultHandle[parametersCount];
+                for (int i = 0; i < parametersCount; i++) {
+                    // the only method argument of io.smallrye.reactive.messaging.Invoker is an object array so we need to pull out
+                    // each argument and put it in the target method arguments array
+                    args[i] = invoke.readArrayValue(invoke.getMethodParam(0), i);
+                    argTypes[i] = method.parameters().get(i).name().toString();
+                }
+                ResultHandle result = invoke.invokeVirtualMethod(
+                        MethodDescriptor.ofMethod(method.declaringClass().name().toString(), method.name(),
+                                method.returnType().name().toString(), argTypes),
+                        beanInstanceHandle, args);
+                if (VOID.equals(method.returnType().name())) {
+                    invoke.returnValue(invoke.loadNull());
+                } else {
+                    invoke.returnValue(result);
+                }
+            }
+        }
+
+        return generatedName.replace('/', '.');
+    }
+
+    private static Class<?> load(String className, ClassLoader cl) {
+        switch (className) {
+            case "boolean":
+                return boolean.class;
+            case "byte":
+                return byte.class;
+            case "short":
+                return short.class;
+            case "int":
+                return int.class;
+            case "long":
+                return long.class;
+            case "float":
+                return float.class;
+            case "double":
+                return double.class;
+            case "char":
+                return char.class;
+            case "void":
+                return void.class;
+        }
+        try {
+            return Class.forName(className, false, cl);
```

Comment:
Should the TTCL be used if not `null`?

Reply:
I am a little sceptical about it. I would prefer to have it explicit to ensure we don't end up with any weird classloader issues. WDYT?

Reply:
Yes, you are right. It may actually open the pandora box.